python_modules/dagster-graphql/dagster_graphql/schema/asset_health.py

```diff
@@ -0,0 +1,78 @@
+import graphene
+
+from dagster_graphql.schema.util import ResolveInfo
+
+
+class GrapheneAssetHealthStatus(graphene.Enum):
+    HEALTHY = "HEALTHY"
+    WARNING = "WARNING"
+    DEGRADED = "DEGRADED"
+    UNKNOWN = "UNKNOWN"
+
+    class Meta:
+        name = "AssetHealthStatus"
+
+
+class GrapheneAssetHealth(graphene.ObjectType):
+    assetHealth = graphene.NonNull(GrapheneAssetHealthStatus)
+    materializationStatus = graphene.Field(GrapheneAssetHealthStatus)
+    assetChecksStatus = graphene.Field(GrapheneAssetHealthStatus)
+    freshnessStatus = graphene.Field(GrapheneAssetHealthStatus)
+
+    class Meta:
+        name = "AssetHealth"
+
+    def resolve_assetHealth(self, graphene_info: ResolveInfo):
+        if not graphene_info.context.instance.asset_health_supported():
```

Comment:
first stab at demonstrating how each indicator status combines into the overall status. We may want to break this down to be more complex if, for example, we want a degraded check to only contribute to a warning status (not a realistic example, just demonstration)